src/annotator/anchoring/text-range.ts

```diff
@@ -32,33 +28,37 @@ function previousSiblingsTextLength(node) {
 }
 
 /**
- * Resolve one or more character offsets within an element to (text node, position)
- * pairs.
+ * Resolve one or more character offsets within an element to (text node,
+ * position) pairs.
  *
- * @param {Element} element
- * @param {number[]} offsets - Offsets, which must be sorted in ascending order
- * @return {{ node: Text, offset: number }[]}
+ * @param element
+ * @param offsets - Offsets, which must be sorted in ascending order
+ * @throws {RangeError}
  */
-function resolveOffsets(element, ...offsets) {
+function resolveOffsets(
+  element: Element,
+  ...offsets: number[]
+): { node: Text; offset: number }[] {
   let nextOffset = offsets.shift();
-  const nodeIter = /** @type {Document} */ (
-    element.ownerDocument
-  ).createNodeIterator(element, NodeFilter.SHOW_TEXT);
+  const nodeIter = element.ownerDocument.createNodeIterator(
+    element,
+    NodeFilter.SHOW_TEXT
+  );
   const results = [];
 
-  let currentNode = nodeIter.nextNode();
+  let currentNode = nodeIter.nextNode() as Text;
```

Comment:
I did poke around to see if there was anyway to type a `NodeIterator` returned by `createNodeIterator` based on args but didn't see anything right away.

Reply:
This case is not correct because `nextNode()` may return null, if the last node has been reached.

Reply:
> This case is not correct because `nextNode()` may return null, if the last node has been reached.

Ah, yes, I caught that in a couple of other cases but not here.